tasks/Task.cpp

```diff
@@ -2,19 +2,57 @@
 
 #include "Task.hpp"
 
+#include <gps_base/BaseTypes.hpp>
+#include <gps_base/rtcm3.hpp>
+
 using namespace gps_ntrip;
+using namespace std;
+
+constexpr int PACKETIZER_BUFFER_SIZE = 4096;
+
+struct RTCMPacketizer : public iodrivers_base::Driver {
+    RTCMPacketizer()
+        : iodrivers_base::Driver(PACKETIZER_BUFFER_SIZE) {}
+
+    int extractPacket(uint8_t const* buffer, size_t size) const override {
+        return gps_base::rtcm3::extractPacket(buffer, size);
+    }
+};
 
 Task::Task(std::string const& name)
-    : TaskBase(name)
+: TaskBase(name)
 {
+    mRTCMPacketizer.reset(new RTCMPacketizer);
+    mRTCMPacketizer->openURI("test://");
+    mRTCMPacketizerStream = dynamic_cast<iodrivers_base::TestStream*>(
+        mRTCMPacketizer->getMainStream()
+    );
+
+    mNTRIP.set_report_interval(10);
+    mNTRIP.OnReceived([this] (const char *buffer, int size) {
+        uint8_t forwardBuffer[4096];
+
+        mRTCMPacketizerStream->pushDataToDriver(vector<uint8_t>(buffer, buffer + size));
+        try {
+            while (true) {
+                size_t size = mRTCMPacketizer->readPacket(
+                    forwardBuffer, PACKETIZER_BUFFER_SIZE, base::Time()
+                );
+
+                iodrivers_base::RawPacket packet;
+                packet.time = base::Time::now();
+                packet.data.insert(packet.data.end(), forwardBuffer, forwardBuffer + size);
+                _rtcm_out.write(packet);
+            }
+        }
+        catch(iodrivers_base::TimeoutError&) {}
+    });
 }
 
 Task::~Task()
 {
 }
```

Comment:
just as curiosity, why do we have to specify the dummy destructor?

Reply:
Because it's code-generated and it really does not hurt to leave it here.